client/Makefile

```diff
@@ -12,7 +12,14 @@ rebuild: clean all
 
 client-kv: client-kv.c
 	cd ../lib/marshal && make
-	$(CC) $(CFLAGS) client-kv.c -o $(OBJS) $(DEPS)
+	$(CC) $(CFLAGS) client-kv.c -o client-kv $(DEPS)
+
+client_api: client_api.c
+	$(CC) -c client_api.c -o client_api.o 
+
+client_api_test: client_api_test.c
+	make client_api && cd ../lib/marshal && make
```

Comment:
First statement is unnecessary (you can assume that `make` is run before `make test`)